kiwi/system/result.py

```diff
@@ -70,6 +74,7 @@ def add(
         :param bool shasum: create shasum when bundling true|false
         """
         if key and filename:
+            self._verify_build_constraints(filename, key)
```

Comment:
Oh, thanks David for pointing this out. I was too fast in my review here. So you are right this implementation will add the size constraint check for any result.add() call. The problem here this would be out of context. Thus I suggest you move _verify_build_constraints() into its own public method of the Result class and give it a meaning, e.g

```python
def verify_image_size(filename, size_limit):
     # compare size_limit with the given filename, raise on size violation
```

The actual call of the method should be done with a context. Meaning you need to look at kiwi/builder and there to all calls of ```self.result.add(...)``` You will find e.g kiwi/builder/container.py

```
self.result.add(
            key='container',
            filename=self.filename,
            use_for_bundle=True,
            compress=False,
            shasum=True
        )
```

For this result file we want to run the size checker. Thus an additional line for every result where we want to run the check is needed. The reading of the correct size constraint from the config section also imho belongs to where the context is. That way we can validate different sizes at the place in the code where all information is present

```python
self.result.verify_image_size(self.filename, self.runtime_config.get_container_image_size_constraint())
```

if the size check is not enabled in the config file, this validation always passes. I think we would need
extra size constraint methods in the config for

* container
* disk
* filesystem
* archive
* livemedia
* installmedia

That should cover all we can build, and in a first shot this does not have to be complete

In order to make the size information part of the result type we need to add a field in the ```result_file_type```
tuple and allow to set that field in the add() method. This imho can be done in a subsequent pull request

Thoughts ?